dbt/adapters/glue/gluedbapi/connection.py

```diff
@@ -382,7 +359,48 @@ def execute(cls,sql,output=True):
             for f in df.schema:
                 d[f.name] = record[f.name]
             results.append(({"type": "record", "data": d}))
-        dumped_results = json.dumps({"type": "results", "rowcount": rowcount,"results": results,"description": [{"name":f.name, "type":str(f.dataType)} for f in df.schema]},default=str)
+        description = [{"name":f.name, "type":str(f.dataType)} for f in df.schema]
+        raw_results = {"type": "results", "rowcount": rowcount, "results": results, "description": description}
+
+        if use_arrow:
+            s3_client = boto3.client('s3')
+            glue_client = boto3.client('glue')
+            security_config_name = glue_client.get_session(Id=session_id)["Session"].get("SecurityConfiguration", None)
+
+            o = urllib.parse.urlparse(location)
+            result_bucket = o.netloc
+            key = urllib.parse.unquote(o.path)[1:]
+            if not key.endswith('/'):
+                key = key + '/'
+            result_key = key + f"tmp-results/{session_id}.feather"
```

Comment:
If the user runs multiple dbt jobs concurrently with reusing the same session, is there any possibility to occur data race? If yes, we can add random ID to the key.

Reply:
That's good catch, I will add the random ID as suffix.

Reply:
Added